src/DynamoCoreWpf/ViewModels/Watch3D/HelixWatch3DViewModel.cs

```diff
@@ -1151,6 +1153,23 @@ private void AggregateRenderPackages(IEnumerable<HelixRenderPackage> packages)
             }
         }
 
+        void pointGeometry3D_MouseDown3D(object sender, RoutedEventArgs e)
+        {
+            var args = e as Mouse3DEventArgs;
+            if (args == null) return;
+            if (args.Viewport == null) return;
+
+            var vm = viewModel as DynamoViewModel;
+            foreach (var node in vm.Model.CurrentWorkspace.Nodes)
+            {
+                var foundNode = node.AstIdentifierBase.Contains(
+                    ((PointGeometryModel3D) e.OriginalSource).Name);
+                if (!foundNode) continue;
+                DynamoSelection.Instance.ClearSelection();
+                vm.Model.AddToSelection(node);
```

Comment:
Although this works, perhaps `DynamoSelection.Instance.ClearSelection();` should be outside of the loop? If we ever anticipate just one node, then we should probably `break` out of the loop once found.